.github/workflows/build-and-deploy-images.yml

```diff
@@ -40,25 +42,32 @@ jobs:
     steps:
       - uses: actions/checkout@v2
 
-      - name: Establish build version environment variable
-        run: echo "BUILD_VERSION=$(cat version_number.txt).$GITHUB_RUN_NUMBER" >> "$GITHUB_ENV"
+      - name: Set BUILD_VERSION for Staging
+        if: github.ref == 'refs/heads/staging'
+        run: echo "BUILD_VERSION=staging-$GITHUB_SHA" >> "$GITHUB_ENV"
+        run: echo "BUILD_VERSION_LATEST=staging" >> "$GITHUB_ENV"
+
+      - name: Set BUILD_VERSION for Production from git tag
+        if: contains(github.ref, 'refs/tags/')
```

Comment:
I suggest adding a `v` at the end of the search string here, so that we're searching for `refs/tags/v`. Because that's exactly the number of characters we're stripping. And it will help provide a small amount of future-proofing if we ever change the `on: tags:` requirement to omit the `v` and change the tag we use to `releases/1.8` or something. I'd rather have the workflow fail to run than have it push something tagged as `eleases/1.8`, missing the initial `r`.

Also, [there's a `startsWith` function](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#startswith), so that's better than `contains` here.

Reply:
done.  Thanks for the GIT_TAG catch!

Reply:
I wasn't as confident about using the tag filter syntax.  I realize that we could be triggering builds for any tag that starts with "v", but I'm willing to take that risk for the time being.